spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
 
-Each chain must satisfy following features to create a interchain account:
+The interchain account specification defines the general way to register an interchain account and transfer tx bytes. The counterparty chain is responsible for deserialising and executing the tx bytes, and the sending chain should know how the counterparty chain will handle the tx bytes in advance. Each chain-specific implementation should clearly document how serialization/deserialization of transactions happens and ensure the required packet data format is clearly outlined. 
 
-- New interchain accounts must not conflict with existing ones.
-- Each chain must keep track of which counterparty chain created each new interchain account.
+Each chain must satisfy the following features to create an interchain account:
 
-Also, each chain must know how the counterparty chains serialise/deserialise transaction bytes in order to send transactions via IBC. And the counterparty chain must implement the process of safely executing IBC transactions by verifying the authority of the transaction's signers.
+- New interchain accounts must not conflict with existing ones
+- Each chain must keep track of which counterparty chain created each new interchain account
 
 The chain must reject the transaction and must not make a state transition in the following cases:
 
-- The IBC transaction fails to be deserialised.
-- The IBC transaction expects signers other than the interchain accounts made by the counterparty chain.
+- The IBC transaction fails to be deserialised
+- The authentication step on the receiving chain (where the interchain account is hosted) fails 
 
-It does not restrict how you can distinguish signers that was not made by the counterparty chain. But the most common way would be to record the account in state when the interchain account is registered and to verify that signers are recorded interchain account.
+#### Known Issues
+##### Ordered Channels 
+In an ordered channel, packet 2 cannot be relayed until packet 1 has been relayed or a valid proof that packet 1 has timed out has been received. If multiple distrusting accounts are allowed to use a single ordered channel this creates an attack vector whereby malicious users can block packets from being received in a timely manner. An example of this may be a malicious user sending hundreds of packets that are never paid for. Other users trying to use this channel will have to wait for all of these packets to be fully relayed or receive valid proof of these packets having timed out before their own packets can be relayed. Therefore, multiplexing N interchain accounts on 1 ordered channel is not viable. This specification assumes the following: N interchain accounts will exist on N ordered channels. 
```

Comment:
```suggestion
In an ordered channel, the next packet cannot be relayed until the previous packet has been relayed. If the previous packet is timed out, the ordered channel will be closed and the next packet can never be relayed. If multiple distrusting accounts are allowed to use a single ordered channel this creates an attack vector whereby malicious users can block packets from being received in a timely manner. An example of this may be a malicious user sending hundreds of packets that are never paid for. Other users trying to use this channel will have to wait for all of these packets to be fully relayed or receive valid proof of these packets having timed out before their own packets can be relayed. Therefore, multiplexing N interchain accounts on 1 ordered channel is not viable. This specification assumes the following: N interchain accounts will exist on N ordered channels. 
```